### PR TITLE
Run verify spelling with make rules

### PR DESCRIPTION
**Pull request by chaodaiG**

### CODE REVIEW

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by chaodaiG** [comment]
Merge conflict resolved, @spiffxp another lgtm please

File: Makefile [hunk 1]
```diff
@@ -70,5 +70,9 @@ update-file-perms:
 	hack/make-rules/update/file-perms.sh
 verify-file-perms:
 	hack/make-rules/verify/file-perms.sh
+update-spelling:
+	hack/make-rules/update/misspell.sh
+verify-spelling:
+	hack/make-rules/verify/misspell.sh
```
Comment (BenTheElder) [inline Makefile:76]: these should be under `.PHONY` (same comment as the other PR https://github.com/kubernetes/test-infra/pull/24213#discussion_r741517101)

Comment (chaodaiG) [reply Makefile:76]: done

File: hack/make-rules/verify/misspell.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+echo "Ensuring go version."
+source ./hack/build/setup-go.sh
+
+# build misspell
+echo "Install misspell."
+cd "hack/tools"
+go build -o "$MISSPELL" github.com/client9/misspell/cmd/misspell
+cd "${REPO_ROOT}"
+
+trap 'echo ERROR: found unexpected instance of "Git"hub, use github or GitHub' ERR
```
Comment (BenTheElder) [inline hack/make-rules/verify/misspell.sh:36]: nit: why the quotes around Git here? we've actually found an instance of `Github` (no quotes) right?

File: hack/make-rules/verify/misspell.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    -o -path '*/localdata/*' \
+    -o -path '*/gubernator/*' \
+    -o -path '*/prow/bugzilla/client_test.go' \
+    -o -path './.git/*' \
+    -o -path './_bin/*' \
+    -o -path './_output/*' \
+    -o -path './_artifacts/*' \
+    -o -path './bazel-*/*' \
+    \) -prune \
+    \) -exec grep -Hn 'Git'hub '{}' '+' ; then
```
Comment (BenTheElder) [inline hack/make-rules/verify/misspell.sh:57]: nit: why isn't this just
```suggestion
    \) -exec grep -Hn Github '{}' '+' ; then
```

File: hack/make-rules/verify/misspell.sh (hunk 2 above)
Comment (chaodaiG) [reply hack/make-rules/verify/misspell.sh:36]: yeah the test itself would fail without quotes

File: hack/make-rules/verify/misspell.sh (hunk 3 above)
Comment (chaodaiG) [reply hack/make-rules/verify/misspell.sh:57]: Same reason as above, this fail the test itself

File: hack/make-rules/verify/misspell.sh (hunk 2 above)
Comment (BenTheElder) [reply hack/make-rules/verify/misspell.sh:36]: lol! in kubernetes we use `tac` to solve this 🙃

**Pull request merged**